payment/src/main/java/org/killbill/billing/payment/core/sm/payments/PaymentOperation.java

```diff
@@ -68,17 +68,15 @@ protected PaymentOperation(final GlobalLocker locker,
 
     @Override
     public OperationResult doOperationCallback() throws OperationException {
-        final String pluginName;
         try {
-            pluginName = daoHelper.getPaymentProviderPluginName(true);
             this.plugin = daoHelper.getPaymentPluginApi();
         } catch (final PaymentApiException e) {
             throw convertToUnknownTransactionStatusAndErroredPaymentState(e);
         }
 
         if (paymentStateContext.shouldLockAccountAndDispatch()) {
             // This will already call unwrapExceptionFromDispatchedTask
-            return doOperationCallbackWithDispatchAndAccountLock(pluginName);
+            return doOperationCallbackWithDispatchAndAccountLock(daoHelper.getPluginName());
```

Comment:
Oh i see. i was getting confused because of the naming: `plugin` is really a `PaymentPluginApi` type (which is unrelated with the `pluginName`). 

Nit: we could rename  `plugin` to `paymentPluginApi` to make it ovbvious